src/main/webapp/app/complaints/complaints-for-students/complaints-student-view.component.ts

```diff
@@ -0,0 +1,160 @@
+     */
+    ngOnInit(): void {
+        this.course = this.exercise.course;
+        this.isExamMode = this.exam != undefined;
+        if (this.participation?.id && this.participation.results && this.participation.results.length > 0) {
+            // Make sure results and participation are connected
+            this.result = this.participation.results[0];
+            this.result.participation = this.participation;
+        }
+        if (this.participation.submissions && this.participation.submissions.length > 0) {
+            this.submission = this.participation.submissions[0];
+        }
+        // for normal exercises we track the number of allowed complaints
+        if (this.course?.complaintsEnabled) {
+            this.complaintService.getNumberOfAllowedComplaintsInCourse(this.course!.id!, this.exercise.teamMode).subscribe((allowedComplaints: number) => {
+                this.numberOfAllowedComplaints = allowedComplaints;
+            });
+        }
+        this.loadPotentialComplaint();
+        this.accountService.identity().then((user) => {
+            if (this.participation && this.participation.student && user?.id) {
+                this.isCurrentUserSubmissionAuthor = this.participation.student.id === user.id;
+            }
+        });
+
+        this.timeOfFeedbackRequestValid = this.isTimeOfFeedbackRequestValid();
+        this.timeOfComplaintValid = this.isTimeOfComplaintValid();
+        this.showComplaintsSection = this.isShowComplaintsSection();
+    }
+
+    /**
+     * Sets the complaint if a complaint and a valid result exists
+     */
+    loadPotentialComplaint(): void {
+        if (this.result && this.result.completionDate) {
+            this.complaintService
+                .findBySubmissionId(this.submission.id!)
+                .pipe(filter((res) => !!res.body))
+                .subscribe((res) => {
+                    this.complaint = res.body!;
+                });
+        }
+    }
+
+    /**
+     * This function is used to check whether the student is allowed to submit a complaint or not.
+     * For exams, submitting a complaint is allowed within the Student Review Period, see {@link isWithinExamReviewPeriod}.
+     *
+     * For normal course exercises, submitting a complaint is allowed within one week after the student received the
+     * result. If the result was submitted after the assessment due date or the assessment due date is not set, the completion date of the result is checked. If the result was
+     * submitted before the assessment due date, the assessment due date is checked, as the student can only see the result after the assessment due date.
+     */
+    private isTimeOfComplaintValid(): boolean {
+        if (this.result?.completionDate) {
+            if (this.isExamMode) {
+                return this.isWithinExamReviewPeriod();
+            }
+            return this.isCompletionDateWithinCourseReviewPeriod(this.result.completionDate);
```

Comment:
why do those very similar methods have different signatures? Wouldn't it be more consistent if either both of them or none of them would have an input parameter.